script.js

```diff
@@ -1,4 +1,275 @@
+//     dice1 = rollDice();
+//     dice2 = rollDice();
+//     currentGameMode = "end game";
+//     if (dice1 > dice2) {
+//       combineNumP2 = "" + dice1 + dice2;
+//       myOutputValue = playerTwo + " the number you got is " + combineNumP2;
+//     } else {
+//       combineNumP2 = "" + dice2 + dice1;
+//       myOutputValue = playerTwo + " the number you got is " + combineNumP2;
+//     }
+//   } else if (currentGameMode == "end game") {
+//     if (combineNumP1 > combineNumP2) {
+//       myOutputValue = playerOne + " wins";
+//     } else {
+//       myOutputValue = playerTwo + " wins";
+//     }
+//   }
+//   return myOutputValue;
+// };
+
+// var currentGameMode = "waiting for username";
+// var userName = "";
+// var randomDice1 = "";
+// var randomDice2 = "";
+// console.log("Dice1: " + randomDice1);
+// console.log("Dice2: " + randomDice2);
+// console.log("" + randomDice1 + randomDice2);
+
+// // dice roll function
+// // var rollDice = function () {
+// //   var diceNum = Math.floor(Math.random() * 6) + 1;
+// //   return diceNum;
+// // };
+// var playDiceGame = function (userName, randomDice1) {
+//   var message = "";
+//   randomDice1 = Math.floor(Math.random() * 6) + 1;
+//   randomDice2 = Math.floor(Math.random() * 6) + 1;
+//   console.log("Dice1: " + randomDice1);
+//   console.log("Dice2: " + randomDice2);
+//   console.log("" + randomDice1 + randomDice2);
+//   message =
+//     userName +
+//     ", you rolled " +
+//     randomDice1 +
+//     " for Dice 1 and " +
+//     randomDice2 +
+//     " for Dice 2.<br><br>Choose the order of the dice.<br>Type '1' to choose Dice 1 or '2' to choose Dice 2 as the first digit.";
+//   return message;
+// };
+
+// var main = function (input) {
+//   var myOutputValue = "";
+
+//   if (currentGameMode == "waiting for username") {
+//     if (input == "") {
+//       myOutputValue = "Sorry please enter your username";
+//     } else {
+//       userName = input;
+//       currentGameMode = "dice game";
+//     }
+//   }
+//   if (currentGameMode == "dice game") {
+//     currentGameMode = "dice choice";
+//     return playDiceGame(userName, input);
+//   }
+
+//   if (currentGameMode == "dice choice") {
+//     if (input == "1") {
+//       console.log("" + randomDice1 + randomDice2);
+//       console.log("Dice1: " + randomDice1);
+//       console.log("Dice2: " + randomDice2);
+//       myOutputValue =
+//         userName +
+//         ", you chose Dice 1 first.<br>Your number is " +
+//         randomDice1 +
+//         randomDice2 +
+//         ".<br> It is now Player 2 turn.";
+//     } else {
+//       myOutputValue =
+//         userName +
+//         ", you chose Dice 2 first.<br>Your number is " +
+//         "" +
+//         randomDice2 +
+//         randomDice1 +
+//         ".<br> It is now Player 2 turn.";
+//     }
+//     currentGameMode = "waiting for username";
+//   }
+//   return myOutputValue;
+// };
+
+var currentGameMode = "waiting for p1 username";
```

Comment:
It seems like you have a few standard gameModes. 

It would be clearer if you delcare all the game modes as globals. Reason being that you want to only have a few standard game modes for clarity. 

e.g. 
// declare game modes
var GAME_MODE_PRE_GAME = '''GAME_MODE_PRE_GAME '
var GAME_MODE_GAME_PLAY = '''GAME_MODE_GAME_PLAY '
var GAME_MODE_USER_SCORE = '''GAME_MODE_USER_SCORE ' 
// set the starting game mode
var currentGameMode = GAME_MODE_PRE_GAME ;

your gamemode conditionals would then look something like this 
if (currentGameMode  == GAME_MODE_PRE_GAME ) {} 

Let me know if this makes sense :) !